src/gt4py/next/program_processors/codegens/gtfn/itir_to_gtfn_ir.py

```diff
@@ -191,6 +191,16 @@ def _literal_as_integral_constant(node: itir.Literal) -> IntegralConstant:
     return IntegralConstant(value=int(node.value))
 
 
+def _is_scan(node: itir.Node) -> bool:
```

Comment:
Why not a `Typeguard[itir.Node]`?

Reply:
upgraded to TypeGuard and which allowed to remove an assert